### PR TITLE
CommCarePlatform and CommCareApp refactor

### PR DESCRIPTION
**Pull request by phillipm**
Misc cleanup of platform and app classes, plus a few other small changes.

Mostly dead code removal and linting.

### CODE REVIEW

File: app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java [hunk 1]
```diff
@@ -716,9 +716,6 @@ else if(resultCode == RESULT_OK) {
                     break;
                 } else if(resultCode == RESULT_OK) {
                     currentState.getSession().setDatum(currentState.getSession().getNeededDatum().getDataId(), intent.getStringExtra(SessionFrame.STATE_DATUM_VAL));
-                    if(intent.hasExtra(CallOutActivity.CALL_DURATION)) {
-                        platform.setCallDuration(intent.getLongExtra(CallOutActivity.CALL_DURATION, 0));
-                    }
```
Comment (dcluna) [inline app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java]: What was this used for before?

Comment (phillipm) [inline app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java]: Oh, for some reason I thought this was unused, but now double checking, it isn't! Thanks for asking, I'll add it back

File: app/src/org/commcare/android/util/AndroidCommCarePlatform.java [hunk 2]
```diff
@@ -96,34 +87,10 @@ public void setProfile(Profile p) {
...
-        }
-    }
-
-    public void setCallDuration(long callDuration) {
-        this.callDuration = callDuration;
-    }
-
-    public long getCallDuration() {
-        return callDuration;
-    }
```
Comment (phillipm) [inline app/src/org/commcare/android/util/AndroidCommCarePlatform.java]: This also needs to be added back in.

File: app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java (hunk 1 above)
Comment (phillipm) [inline app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java]: @dcluna Actually, that line calls AndroidCommCarePlatform.setCallDuration, which sets a class field that is never accessed. That is why I took all this out. The functionality seems to have been moved to CalloutActivity.java. Please double check this (that AndroidCommCarePlatform.getCallDuration is never called), but I think what this PR removes is never used.

Comment (phillipm) [inline app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java]: According to @ctsims, all the call duration logic can be removed, so we shouldn't have to worry to hard about taking out something important

**Pull request merged**